docs/docs/20-concepts/02-mcp-gateway/03-obot-registry.md

```diff
@@ -0,0 +1,288 @@
+
+### Case 1: Fixed Endpoint URL
+
+Use this when your server has one static URL for all users:
+
+```yaml
+runtime: remote
+remoteConfig:
+  fixedURL: https://api.example.com/v1/mcp
+  headers:
+    - name: Personal Access Token
+      description: PAT
+      key: Authorization           # HTTP header name
+      required: true
+      sensitive: true
+
+```
+
+### Case 2: Same Hostname, User Selects Path
+
+Use this when users connect to different paths on your domain:
+
+```yaml
+runtime: remote
+remoteConfig:
+  hostname: api.example.com
+  headers:
+    - name: API Key
+      description: Your API key from <https://example.com/settings>
+      key: X-API-Key
+      required: true
+      sensitive: true
+
+```
+
+Users will specify their path when connecting (e.g., https://api.example.com/serviceA/mcp).
+
+### Case 3: URL Built from User Environment
+
+Use this when the URL includes user-specific values:
+
+```yaml
+env:
+  - key: WORKSPACE_URL
+    name: Workspace URL
+    description: "Your workspace URL, e.g., <https://mycompany.cloud.com>"
+    required: true
+    sensitive: false
+
+runtime: remote
+remoteConfig:
+  URLTemplate: ${WORKSPACE_URL}/api/2.0/mcp/
+  headers:
+    - name: Personal Access Token
+      description: PAT with workspace access
+      key: Authorization
+      required: true
+      sensitive: true
+
+```
+
+---
+</details>
+
+<details>
+  <summary>Option B: Containerized HTTP Server</summary>
+
+Use this if you already have a Docker image that serves streamable HTTP:
+
+```yaml
+runtime: containerized
+containerizedConfig:
+  image: your-dockerhub-user/your-mcp-server:v1.0.0
+  port: 8080           # Your container's exposed HTTP port
+  path: /mcp           # HTTP path where MCP endpoint is served
+  args:                # Optional runtime flags
+    - flags # flags needed.
+
+```
+
+**Requirements:**
+
+- Your container must serve HTTP/SSE on the specified port
+- The MCP endpoint must be available at the specified path
+- Image must be publicly accessible (Docker Hub, GHCR, etc.)
+
+---
+</details>
+
+
+<details>
+  <summary>Option C: Stdio MCP Server (npx, uvx, docker)</summary>
+
+**⚠️ Important:** If your server is stdio-based, do NOT complete this section yet.
+**Instead:**
+
+1. Complete **Step 2** below to repackage your stdio server
+2. Wait for your image to be published to GHCR
+3. Return here and use this runtime block:
+
+```yaml
+runtime: containerized
+containerizedConfig:
+  image: ghcr.io/obot-platform/mcp-images/<your-server-name>:<tag>
+  port: 8099         # Fixed for stdio servers
+  path: /            # Fixed for stdio servers
+  args:
+    - <your-mcp-server-command>           # Command to run your stdio server
+    # Add any additional flags your server needs:
+    # - --region
+    # - us-east-1
+
+```
+
+The `args` must include the command that starts your stdio server, plus any required flags.
+
+---
+</details>
+
+---
+
+---
+
+## Step 2 — (Only for stdio) Repackage and publish an image
```

Comment:
This section should be reworked as well because we require this for docker images as well.

Reply:
reworked, added instructions to repackage http docker image.